### ISSUE TITLE
os.execute dont called as expected.

### ISSUE THREAD

**Issue by friendwu** [premake/premake-core#279]
I'm now using os.execute to retrieve 3rd packages, code like below:

```
        local xx = "wget https://github.com/redis/hiredis/archive/v0.13.3.tar.gz -O ./deps/hiredis_v0.13.3.tar.gz"
        printf(xx)
        local z = os.execute(xx)
        if z ~= 0 then
            error("download pkg failed", 0)
        end 
```

when I run premake5, the url are translated into a wrong value, log message are as below:

```
$ premake5 deps
Running action 'deps'...
getting deps hiredis
wget https://github.com/redis/hiredis/archive/v0.13.3.tar.gz -O ./deps/hiredis_v0.13.3.tar.gz
--2015-09-21 16:20:23--  ftp://https//github.com/redis/hiredis/archive/v0.13.3.tar.gz
           => "./deps/hiredis_v0.13.3.tar.gz"
Resolving https... failed: No address associated with hostname.
wget: unable to resolve host address “https”
Error: download hiredis failed
```

PS: my own lua interpreter can call os.execute as expected:

```
$ lua
Lua 5.2.3  Copyright (C) 1994-2013 Lua.org, PUC-Rio
> require "os"
> os.execute("wget https://github.com/redis/hiredis/archive/v0.13.3.tar.gz -O /vagrant/work/updns/deps/hiredis_v0.13.3.tar.gz")
--2015-09-21 16:22:03--  https://github.com/redis/hiredis/archive/v0.13.3.tar.gz
Resolving github.com... 
```

it really confuse me.

**Comment by friendwu** [comment]
and just now I found premake support http.download, this API is not documented, and when I try it, the response message shows me that https is an Unsupported protocol.

**Comment by tritao** [comment]
I've now documented the HTTP APIs on the wiki. About HTTPS, there's a pending pull request to support it, we should have it soon.

**Comment by samsinsane** [comment]
This bug got me at work, I figured out that it's because premake performs `path.translate` on the string before passing it to the original Lua `os.execute`. Is there any reason why we need to translate the string before executing it? The translate seems unnecessary, and mostly destructive?

**Comment by TurkeyMan** [comment]
Aye, does seem weird...

**Comment by tritao** [comment]
Hmm... @samsinsane... that got me thinking.. that's probably why I've been getting some bugs when invoking commands with HTTP URIs.. they're passed down to the OS missing some slashes...

**Comment by samsinsane** [comment]
@tritao there's a good chance this is the reason, I was actually doing the same thing when I found it. I'm happy to create a PR for this but I'm concerned that there's a need to perform the translate, @starkos any thoughts?

**Comment by starkos** [comment]
No, I agree, we can't translate the entirety of a command line. Thanks for adding the docs @tritao!

**Comment by tvandijck** [comment]
```lua
premake.override(os, "execute", function(base, cmd)
		cmd = path.normalize(cmd)
		cmd = os.translateCommands(cmd)
		return base(cmd)
	end)
```
translateCommands is a very different thing, it's not a path.translate.

The path.normalize used to only normalize the string up to the first space.. I'm not sure if we have a test for that, if not there should be... the path.normalize has changes quite a bit since I last touched and looked at it, so I don't know if that behavior is still maintained..

basically... 
- path.normalize should only normalize the "command" part of the string up to the first space where the arguments begin...
- os.translateCommands should only expand command line tokens like {COPY} etc..

**Comment by tvandijck** [comment]
looking at the history of path.normalize, I can't actually find the behavior I described... I could have sworn that it worked like that before, but clearly I was wrong... so that makes my point somewhat moot...

that said, I think we can make it work that way for the os.execute case... only normalize up to the first space, but this would have to be done AFTER the os.translateCommands

**Comment by samsinsane** [comment]
Sorry about that, I meant `path.normalize`, I made the same mistake while trying to track down the bug lol.

**Comment by samsinsane** [comment]
Also, I have the same recollection of `path.normalize` being like that, maybe it was a pending PR? I'm going to assume it was closed because it didn't handle commands with spaces in them, for example `"path with spaces/binary" args`, and I think you can do something like `path\ with\ spaces/binary args`?

**Comment by tritao** [comment]
This is still an issue with alpha 8.

**Comment by nmoehrle** [comment]
Should be resolved through #570

**Comment by samsinsane** [comment]
Closing based on above comment, if the issue is still occurring please reopen.
